<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>478</width>
    <height>493</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenCV StereoCorrespondenceBM Tuner</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QLabel" name="label_image_right">
        <property name="maximumSize">
         <size>
          <width>480</width>
          <height>360</height>
         </size>
        </property>
        <property name="text">
         <string>Right image</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_depth_map">
        <property name="text">
         <string>                              Depth_map</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <layout class="QGridLayout" name="gridLayout_buttons">
        <item row="12" column="2">
         <widget class="QSlider" name="horizontalSlider_uniqueness_ratio">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>15</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <widget class="QLabel" name="label_P2">
          <property name="text">
           <string>P2</string>
          </property>
         </widget>
        </item>
        <item row="12" column="0">
         <widget class="QLabel" name="label_uniqueness_ratio">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Margin in percentage by which the best (minimum) computed cost function value should “win” the second best value to consider the found match correct. Normally, a value within the 5-15 range is good enough.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Uniqueness ratio</string>
          </property>
         </widget>
        </item>
        <item row="13" column="2">
         <widget class="QSlider" name="horizontalSlider_speckle_window_size">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="7" column="2">
         <widget class="QSlider" name="horizontalSlider_disp_12_max_diff">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="minimum">
           <number>-1</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>-1</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="8" column="2">
         <widget class="QSlider" name="horizontalSlider_pre_filter_cap">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>63</number>
          </property>
          <property name="value">
           <number>31</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSlider" name="horizontalSlider_min_disparity">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="minimum">
           <number>-300</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="pageStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>-64</number>
          </property>
          <property name="tracking">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QSlider" name="horizontalSlider_P1">
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_SAD_window_size">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Matched block size. It must be an odd number &lt;span style=&quot; font-family:'Courier New,courier';&quot;&gt;&amp;gt;=1&lt;/span&gt; . Normally, it should be somewhere in the &lt;span style=&quot; font-family:'Courier New,courier';&quot;&gt;3..11&lt;/span&gt; range.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Block size</string>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QSlider" name="horizontalSlider_block_size">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
          <property name="value">
           <number>41</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="14" column="2">
         <widget class="QSlider" name="horizontalSlider_speckle_range">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>3</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="label_pre_filter_cap">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Truncation value for the prefiltered image pixels. The algorithm first computes x-derivative at each pixel and clips its value by &lt;span style=&quot; font-family:'Courier New,courier';&quot;&gt;[-preFilterCap, preFilterCap]&lt;/span&gt; interval. The result values are passed to the Birchfield-Tomasi pixel cost function.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Pre-filter cap</string>
          </property>
         </widget>
        </item>
        <item row="8" column="3">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>63</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="label_0">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="13" column="3">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>100</string>
          </property>
         </widget>
        </item>
        <item row="14" column="3">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>32</string>
          </property>
         </widget>
        </item>
        <item row="5" column="3">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>216</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QSlider" name="horizontalSlider_num_of_disparity">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="minimum">
           <number>16</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="singleStep">
           <number>16</number>
          </property>
          <property name="pageStep">
           <number>16</number>
          </property>
          <property name="value">
           <number>128</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="6" column="3">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>864</string>
          </property>
         </widget>
        </item>
        <item row="12" column="3">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>10</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QLabel" name="label_min_disparity">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Minimum possible disparity value. Normally, it is zero but sometimes rectification algorithms can shift images, so this parameter needs to be adjusted accordingly.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Minimum disparity</string>
          </property>
         </widget>
        </item>
        <item row="14" column="0">
         <widget class="QLabel" name="label_speckle_range">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum disparity variation within each connected component. If you do speckle filtering, set the parameter to a positive value, it will be implicitly multiplied by 16. Normally, 1 or 2 is good enough.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Speckle range</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="label_disp_12_max_diff">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum allowed difference (in integer pixel units) in the left-right disparity check. Set it to a non-positive value to disable the check.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Disp. 12 max. diff.</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_num_of_disparity">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum disparity minus minimum disparity. The value is always greater than zero. In the current implementation, this parameter must be divisible by 16.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Number of disparity</string>
          </property>
         </widget>
        </item>
        <item row="13" column="0">
         <widget class="QLabel" name="label_speckle_window_size">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum size of smooth disparity regions to consider their noise speckles and invalidate. Set it to 0 to disable speckle filtering. Otherwise, set it somewhere in the 50-200 range.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Speckle window size</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_P1">
          <property name="midLineWidth">
           <number>960</number>
          </property>
          <property name="text">
           <string>P1</string>
          </property>
         </widget>
        </item>
        <item row="4" column="3">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>3</string>
          </property>
         </widget>
        </item>
        <item row="6" column="2">
         <widget class="QSlider" name="horizontalSlider_P2">
          <property name="maximumSize">
           <size>
            <width>275</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QLabel" name="label_1">
          <property name="text">
           <string>144</string>
          </property>
         </widget>
        </item>
        <item row="7" column="3">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>1</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_image_left">
        <property name="maximumSize">
         <size>
          <width>480</width>
          <height>360</height>
         </size>
        </property>
        <property name="text">
         <string>Left image</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton_right">
        <property name="text">
         <string>Load right image</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pushButton_left">
        <property name="text">
         <string>Load left image</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
